pygments/lexers/theorem.py

```diff
@@ -144,8 +147,26 @@ class CoqLexer(RegexLexer):
             (r'[~?][a-z][\w\']*:', Name),
             (r'\S', Name.Builtin.Pseudo),
         ],
+        'set-options': [
+            (r'\s+', Text),
+            (r'[A-Z]\w*', Keyword.Namespace),
+            (r'"', String.Double, 'string'),
+            (r'\d+', Number.Integer),
+            (r'.', Punctuation, '#pop'),
+        ],
+        'sn-notation': [
+            (r'\s+', Text),
+            # Extra keywords to highlight only in this scope
+            (r'\b(?:via|mapping|abstract|warning|after)\b', Keyword),
+            (r'=>|[()\[\]:,]', Operator),
+            (r'\b[^\W\d][\w\']*(?:\.[^\W\d][\w\']*)*\b', Name),
+            (r'\d[\d_]*', Number.Integer),
+            (r'0[xX][\da-fA-F][\da-fA-F_]*', Number.Hex),
+            (r'\(\*', Comment, 'comment'),
+            (r'\.', Punctuation, '#pop'),
+        ],
         'comment': [
-            (r'[^(*)]+', Comment),
+            (r'([^(*)]+|\*+(?!\)))+', Comment),
```

Comment:
I guess this is to produce fewer tokens on "heading" comments like `(************************)`? Could you add a comment above this line to explain it (so someone else doesn't remove this to simplify the code)?